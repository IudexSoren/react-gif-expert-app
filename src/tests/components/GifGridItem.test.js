import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';
import '@testing-library/jest-dom';

describe('Pruebas de GifGridItem', () => {

  const title = 'Title de prueba';
  const url = 'Url de prueba';
  const wrapper = shallow(<GifGridItem title={ title } url={ url } />);

  test('Debe renderizar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe tener un párrafo con el título', () => {
    const titulo = wrapper.find('p').text();

    expect(titulo).toBe(title);
  });

  test('Debe tener una imagen con los atributos src y alt', () => {
    const img = wrapper.find('img');
    const { src, alt } = img.props();

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Debe tener la clase animate__fadeInUp', () => {
    const card = wrapper.find('.card');

    expect(card.hasClass('animate__fadeInUp')).toBe(true);
  })


})
