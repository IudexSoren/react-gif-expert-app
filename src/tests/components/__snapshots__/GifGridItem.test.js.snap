// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGridItem Debe renderizar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <img
    alt="Title de prueba"
    src="Url de prueba"
  />
  <p>
    Title de prueba
  </p>
</div>
`;
